frontend/src/app/pages/source-detail/source-detail.component.ts

```diff
@@ -64,17 +64,18 @@ export class SourceDetailComponent implements OnInit {
   }
   getPatientDOB(){
     return getPath(this.selectedPatient?.resource_raw, 'birthDate')
-
+  }
+  getPatientAge(){
+    var msInYear = 365 * 24 * 60 * 60 * 1000;
+    var age = Date.now() - getPatientDOB();
+    return Math.floor(age / msInYear);
```

Comment:
This function should gracefully handle invalid/null values in `getPatientDOB()`

getPatientDOB may be null, a String or a Date.